deriva/core/ermrest_config.py

```diff
@@ -139,17 +139,29 @@ class NodeConfig (object):
          self.generated: treat tag.generated as a boolean
          self.immutable: treat tag.immutable as a boolean
     """
-    def __init__(self, uri_path, node_doc):
-        self.uri_path = uri_path
-        self.update_uri_path = uri_path
+    def __init__(self, node_doc):
         self.annotations = dict(node_doc.get('annotations', {}))
         self._supports_comment = True
         self._comment = node_doc.get('comment')
 
-    def apply(self, catalog, existing=None):
+    @property
+    def catalog(self):
+        raise NotImplementedError('derived classes MUST define self.catalog property')
```

Comment:
could make it an `abc.abstractproperty` possibly, if its required to be defined by the derived classes